e2e/framework/util.go

```diff
@@ -32,3 +34,13 @@ func GetControlPlaneHostAndPort(cl client.Client) (string, int32, error) {
 
 	return apiUrl.Hostname(), int32(port), nil
 }
+
+// IsTechPreviewNoUpgrade checks if a cluster is a TechPreviewNoUpgrade cluster
```

Comment:
You should check for the presence of a particular feature gate being enabled in the feature gate status rather than checking tech preview directly.

If we can use the openshift tests extension (I think we got that?), then naming tests based on the feature gate name will be sufficient and the test extension will skip any tests that aren't enabled for us

I realise we don't have a great feature gate for this feature right now, I'll take an action item to create that so that we can start moving on with this